internal/agenthttp/client.go

```diff
@@ -0,0 +1,136 @@
+// Package agenthttp creates standard Go [net/http.Client]s with common config
+// options.
+package agenthttp
+
+import (
+	"crypto/tls"
+	"net/http"
+	"sync"
+	"time"
+
+	"golang.org/x/net/http2"
+)
+
+// NewClient creates a HTTP client.
+func NewClient(opts ...ClientOption) *http.Client {
+	conf := clientConfig{
+		// This spells out the defaults, even if some of them are zero values.
+		Bearer:     "",
+		Token:      "",
+		AllowHTTP2: true,
+		Timeout:    60 * time.Second,
+		TLSConfig:  nil,
+	}
+	for _, opt := range opts {
+		opt(&conf)
+	}
+
+	cacheKey := transportCacheKey{
+		AllowHTTP2: conf.AllowHTTP2,
+		TLSConfig:  conf.TLSConfig,
+	}
+
+	transportCacheMu.Lock()
+	transport := transportCache[cacheKey]
+	if transport == nil {
+		transport = newTransport(&conf)
+		transportCache[cacheKey] = transport
+	}
+	transportCacheMu.Unlock()
```

Comment:
Is the goal by sharing the transports here to also share the connection pooling?

Once side effect is when we call `CloseIdleConnections` that impacts all clients using this transport, even if they have a different http client. I think that's ok?

Reply:
Yes, that's the idea, but I expect it will have very limited impact.